H1 thickline: Fix border rendering bug in older browsers

EricDunsworth opened the pull request:
Some older browsers (most notably Safari <= 15.3) incorrectly set ``border-style: solid;`` whenever the ``border-image`` property was used. That in turn caused affected browsers to render the thickline style as a three-sided border.

This works around the issue by setting the thickline style's left/right/top ``border-*-width`` properties to 0.

Thanks to @donmcdill for pitching this approach (other ways I prototyped didn't work).

Fixes #2396.

EricDunsworth commented:
Just made a few minor updates:
* Mentioned "border" at the start of the commit message (old wording felt too ambiguous)
* Copied ``sites/main-page-title/index.json-ld``'s changes into ``_data/sites.json``
* Adjusted modified dates to today

duboisp commented:
@Garneauma please check and sync this change with the other PR for the stack header #2412

EricDunsworth commented:
Just rebased to fix the merge conflicts and recommitted with a newer modified date.

Garneauma commented:
Pre-approved upon successful review.

EricDunsworth commented:
Just rebased again and updated modified dates.

Btw this was rebased "as-is" (didn't need to manually-edit anything to resolve the merge conflicts). I didn't change anything to accommodate #2412's additions.

Garneauma reviewed (approved):
@EricDunsworth Thank you for this contribution. Tested in Safari 14 and 13 using Browserstack and the fix works.
